Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift

```diff
@@ -0,0 +1,302 @@
+                                  description: collection.description,
+                                  keywords: collection.keywords,
+                                  packages: packages,
+                                  createdAt: collection.generatedAt,
+                                  lastProcessedAt: Date()))
+        }
+    }
+
+    private func makeRequestOptions(validResponseCodes: [Int]) -> HTTPClientRequest.Options {
+        var options = HTTPClientRequest.Options()
+        options.addUserAgent = true
+        options.validResponseCodes = validResponseCodes
+        options.timeout = self.configuration.requestTimeout
+        // TODO: consider making these configurable
+        options.retryStrategy = .exponentialBackoff(maxAttempts: 3, baseDelay: .milliseconds(50))
+        options.circuitBreakerStrategy = .hostErrors(maxErrors: 5, age: .seconds(5))
+        return options
+    }
+
+    public struct Configuration {
+        public var maximumSizeInBytes: Int
+        public var requestTimeout: DispatchTimeInterval?
+
+        public init(maximumSizeInBytes: Int? = nil,
+                    requestTimeout: DispatchTimeInterval? = nil) {
+            // TODO: where should we read defaults from?
+            self.maximumSizeInBytes = maximumSizeInBytes ?? 5_000_000 // 5MB
+            self.requestTimeout = requestTimeout ?? .seconds(1)
+        }
+    }
+
+    public enum Errors: Error {
+        case invalidJSON(Error)
+        case invalidResponse(String)
+        case responseTooLarge(Int)
+    }
+}
+
+// MARK: - JSON Models
+
+extension JSONPackageCollectionProvider {
+    fileprivate struct CollectionV1: Decodable {
+        let name: String
+        let description: String?
+        let keywords: [String]?
+        let formatVersion: String
+        let revision: Int?
+        let packages: [Package]
+        let generatedAt: Date
+        let generatedBy: Author?
+
+        private enum CodingKeys: String, CodingKey {
+            case name
+            case description
+            case keywords
+            case formatVersion
+            case revision
+            case packages
+            case generatedAt
+            case generatedBy
+        }
+    }
+}
+
+extension JSONPackageCollectionProvider.CollectionV1 {
+    fileprivate struct Package: Decodable {
+        let url: String
+        let description: String?
+        let keywords: [String]?
+        let versions: [Version]
+        let readmeURL: String?
+
+        enum CodingKeys: String, CodingKey {
+            case url
+            case description
+            case keywords
+            case versions
+            case readmeURL
+        }
+    }
+
+    fileprivate struct Version: Decodable {
+        let version: String
+        let packageName: String
+        let toolsVersion: String
+        let targets: [Target]
+        let products: [Product]
+        let minimumPlatformVersions: [PlatformVersion]?
```

Comment:
Need to add this to the collection version model and the parsing from JSON to model

Reply:
I can do that later when I update the JSON per https://forums.swift.org/t/package-collection-format/42071